extensions/binding/org.eclipse.smarthome.binding.digitalstrom/src/main/java/org/eclipse/smarthome/binding/digitalstrom/internal/DigitalSTROMHandlerFactory.java

```diff
@@ -0,0 +1,259 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.binding.digitalstrom.internal;
+
+import static org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.smarthome.binding.digitalstrom.DigitalSTROMBindingConstants;
+import org.eclipse.smarthome.binding.digitalstrom.handler.DeviceHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.SceneHandler;
+import org.eclipse.smarthome.binding.digitalstrom.handler.BridgeHandler;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.SceneDiscoveryService;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.DeviceDiscoveryService;
+import org.eclipse.smarthome.binding.digitalstrom.internal.discovery.SceneDiscoveryServiceManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.ConnectionManager;
+import org.eclipse.smarthome.binding.digitalstrom.internal.lib.manager.impl.ConnectionManagerImpl;
+import org.eclipse.smarthome.binding.digitalstrom.internal.providers.DsThingTypeProvider;
+import org.eclipse.smarthome.config.core.Configuration;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingTypeProvider;
+import org.osgi.framework.ServiceRegistration;
+
+import com.google.common.collect.Sets;
+
+/**
+ * The {@link DigitalSTROMHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Michael Ochel - Initial contribution
+ * @author Mathias Siegele - Initial contribution
+ *
+ */
+public class DigitalSTROMHandlerFactory extends BaseThingHandlerFactory {
+
+    private Map<String, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = Sets.union(SceneHandler.SUPPORTED_THING_TYPES,
+            Sets.union(BridgeHandler.SUPPORTED_THING_TYPES, DeviceHandler.SUPPORTED_THING_TYPES));
+
+    private ConnectionManager connMan = null;
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES.contains(thingTypeUID);
```

Comment:
We tested the channel creation for the GenericDeviceTypes by deleting one channel from the ThingType and add it later through the Handler to a Thing. This is working right, the channel was created with the right ChannelTypeUID etc. and is added to the Thing.

So far so good, but if we/someone want/s to configure the Channel (link a Item) at the PaperUI from openHAB2, the Channel won’t shown at the Configuration/Things/exampleThing. We think, that is because the PaperUI currently map a new added Channel only through the ChannelId to an existing Channel with the same id of the ThingType (like we do it now) and not through the ChannelTypeUID to a ChannelDefiniton.

We think it is better solution to add the Channels only, if we know which channels are really needed. Thus a ThingType will be created without Channels and we add them later. But then the configuration by the PaperUI is not really usefull till the PaperUI haven’t implemented the mapping of ChannelDefinitions through the ChannelTypeUID. 

The other way would be to create the ThingType with all possible Channels and delete the unused later. Than the configuration by the PaperUI is useful, but it will throw WARN logger messages because the Channels can’t be found.

What's the better solution in your opinion?